lexos/static/js/scripts_cut.js

```diff
@@ -1,277 +1,401 @@
-    footerButtons = '<button type="button" class="btn btn-default" id="warningContinue">Continue Anyway</button>'
-    footerButtons += '<button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>'
+    const sizeWarning = `Current cut settings will result in over
+    100 new segments. Please be patient if you continue.`
+    const footerButtons = `<button type="button" class="btn btn-default" id="warningContinue">Continue Anyway</button>
+    <button type="button" class="btn btn-default" data-dismiss="modal">Cancel</button>`
     $('#warning-modal-footer').html(footerButtons)
     $('#warning-modal-message').html(sizeWarning)
     // Hide the processing icon and show the modal
-    $('#status-prepare').css({ 'visibility': 'hidden' })
+    $('#status-prepare').css({'visibility': 'hidden'})
     $('#warning-modal').modal()
   } else {
     $('#needsWarning').val('false')
   }
 }
 
-var xhr
+let xhr
+
+/**
+ * Performs the ajax request.
+ * @param {string} action - the action type being requested.
+ * @returns {void}
+ */
 function doAjax (action) {
   /* It's not really efficient to create a FormData and a json object,
      but the former is easier to pass to lexos.py functions, and the
      latter is easier for the ajax response to use. */
-  var formData = new FormData($('form')[0])
+  const numActiveFiles = $('#numActiveFiles').val()
+  const formData = new FormData($('form')[0])
   formData.append('action', action)
-  var jsonform = jsonifyForm()
-  $.extend(jsonform, { 'action': action })
+  const jsonForm = jsonifyForm()
+  $.extend(jsonForm, {'action': action})
   // Initiate a timer to allow user to cancel if processing takes too long
-  var loadingTimeout = window.setTimeout(function () {
+  const loadingTimeout = window.setTimeout(function () {
     $('#needsWarning').val('true')
-    var timeWarning = 'Lexos seems to be taking a long time. This may be because you are cutting a large number of documents. If not, we suggest that you cancel, reload the page, and try again.'
-    footerButtons = '<button type="button" class="btn btn-default" data-dismiss="modal">Continue Anyway</button>'
-    footerButtons += '<button type="button" class="btn btn-default" id="timerCancel" >Cancel</button>'
+    const timeWarning = `Lexos seems to be taking a long time.  \
+    This may be because you are cutting a large number of documents. 
+    If not, we suggest that you cancel, reload the page, and try again.`
+    const footerButtons = `<button type="button" class="btn btn-default" data-dismiss="modal">Continue Anyway</button>
+    <button type="button" class="btn btn-default" id="timerCancel" >Cancel</button>`
     $('#warning-modal-footer').html(footerButtons)
     $('#warning-modal-message').html(timeWarning)
     $('#warning-modal').modal()
-  }, 10000) // 10 weconds
+  }, 10000) // 10 seconds
   xhr = $.ajax({
```

Comment:
I think we may don't need ```xhr = ``` for the ajax call. The documentation can be found [here](http://api.jquery.com/jquery.ajax/). Maybe try to remove it and see if it still works? Thanks..

Reply:
Yes still works fine after removing xhr.